Windows: ` $<TARGET_FILE:libtarget-shared>` refers to non-existent file when `CMAKE_LIBRARY_OUTPUT_DIRECTORY` is set

When `CMAKE_LIBRARY_OUTPUT_DIRECTORY` is set, the `TARGET_FILE` generator expression for an imported cdylib library crate points to a non-existent file. This makes it difficult to provide an "install" target that installs generated DLLs.

See the attached zip archive for a reproducer.

Built on Windows with CMake 3.24.2, Ninja, and MSVC:

```
>cmake -DCMAKE_INSTALL_PREFIX=c:\dev\corrosion-bug -GNinja ..
-- The CXX compiler identification is MSVC 19.34.31937.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Rust Toolchain: stable-x86_64-pc-windows-msvc
-- Rust Target: x86_64-pc-windows-msvc
-- Found Rust: C:/Users/Simon Hausmann/.rustup/toolchains/stable-x86_64-pc-windows-msvc/bin/rustc.exe (found version "1.66.1")
-- Using Corrosion as a subdirectory
-- Configuring done
-- Generating done
-- Build files have been written to: C:/dev/corrosion-bug/build

>cmake --build .
...

>cmake --install .
-- Install configuration: "Debug"
CMake Error at cmake_install.cmake:41 (file):
  file INSTALL cannot find "C:/dev/corrosion-bug/build/corrosion_bug.dll":
  File exists.

```

[corrosion-bug-main.zip](https://github.com/corrosion-rs/corrosion/files/10756813/corrosion-bug-main.zip)

(The reproducer is also at https://github.com/tronical/corrosion-bug but I might delete that repo in the future, so archiving as zip here to be on the safe side :)

--- a/cmake/Corrosion.cmake
+++ b/cmake/Corrosion.cmake
@@ -128,7 +128,7 @@ endfunction()
 function(_corrosion_set_imported_location_deferred target_name base_property output_directory_property filename)
     # The output directory property is expected to be set on the exposed target (without postfix),
     # but we need to set the imported location on the actual library target with postfix.
-    if("${target_name}" MATCHES "^([^-]+)-(static|shared)$")
+    if("${target_name}" MATCHES "^(.+)-(static|shared)$")
         set(output_dir_prop_target_name "${CMAKE_MATCH_1}")
     else()
         set(output_dir_prop_target_name "${target_name}")

